src/Glpi/Search/Input/QueryBuilder.php

```diff
@@ -845,9 +848,66 @@ public static function cleanParams(array $params): array
             }
         }
 
+        self::validateCriteria($params);
+
         return $params;
     }
 
+    private static function validateCriteria(array &$params): void
+    {
+        if (!isset($params['criteria'])) {
+            return;
+        }
+        $valid_main_opts = SearchOption::getOptionsForItemtype($params['itemtype']);
+
+        // Validate criteria
+        $invalid_criteria = [];
+        foreach ($params['criteria'] as $k => $criterion) {
+            if (!array_key_exists('field', $criterion) || !is_numeric($criterion['field'])) {
+                continue;
+            }
+            if (isset($criterion['itemtype'])) {
```

Comment:
When `$criterion['itemtype']` is identical to `$params['itemtype']`, we could directly check in `$valid_main_opts`.

```suggestion
            if (isset($criterion['itemtype']) && $criterion['itemtype'] !== $params['itemtype']) {
```